client/src/components/ResetPassword.jsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import AppActions from '../actions/AppActions';
+import Header from '../components/Navbar.jsx';
+/**
+ * @return {void}
+ * @class ResetPassword
+ * @extends {React.Component}
+ */
+class ResetPassword extends React.Component {
+  /**
+   * Creates an instance of ResetPassword.
+   * @param {any} props
+   * @memberof ResetPassword
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      message: ''
+    };
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
+  }
+  /**
+   * Creates an on change event method
+   * @return {void}
+   * @param {any} event
+   * @memberof ResetPassword
+   */
+  onChange(event) {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+  /**
+   * Creates an on submit event that fires a submit action.
+   * @return {void}
+   * @param {any} event
+   * @memberof ResetPassword
+   */
+  onSubmit(event) {
+    event.preventDefault();
+    const resetEmail = {
+      email: this.state.email
+    };
+    AppActions.resetPassword(resetEmail);
```

Comment:
I think you can simply just do `AppActions.resetPassword({ email: this.state.email })`. The main idea of having variables like resetEmail should be for reusability which is not happening in this case

Reply:
Are you trying trying to reset the email or the password? I think you're trying to confirm the user's email before resetting the password. However, there's no parameter that indicates where/how a new password is being used or set.
For instance something like AppActions.resetPassword(email, newPassword) takes two parameters: user's email (which will be verified) and new password(which is to be set).